ts/peer.ts

```diff
@@ -0,0 +1,29 @@
+import PeerId from 'peer-id'
+import { Pushable } from 'it-pushable'
+import { Message, SubOpts } from './message'
+
+// eslint-disable-next-line @typescript-eslint/no-empty-interface
+export interface Connection {}
+
+export interface Peer {
+  id: PeerId
+  protocols: string[]
+  conn: Connection
+  topics: Set<string>
+  stream: Pushable<Buffer>
+  readonly isConnected: boolean
+  readonly isWritable: boolean
+  write (buf: Buffer): void
+  attachConnection (conn: Connection)
+  sendSubscriptions (topics: string[]): void
+  sendUnsubscriptions (topics: string[]): void
+  sendMessages (msgs: Message[]): void
+  updateSubscriptions (subOpts: SubOpts[]): void
+  close (): void
+}
+
+export interface Registrar {
+  handle (): void
+  register (): void
+  unregister (): void
+}
```

Comment:
You should probably update those interfaces in libp2p-ts and use from there

Reply:
I want to upstream these to js-libp2p-pubsub directly :)
Leaving here for now, but will open a PR to add ts declarations there soon.